File: app/src/main/java/com/sp/base/database/JsonTypeConverter.java
package com.sp.base.database;

import androidx.room.TypeConverter;

import java.nio.charset.StandardCharsets;

public class JsonTypeConverter {
    @TypeConverter
    public static byte[] encodeJsonResponse(String value) {
        return value.getBytes();
    }

    @TypeConverter
    public static String decodeJsonResponse(byte[] encodedValue) {
        return new String(encodedValue, StandardCharsets.UTF_8);
    }
}

File: app/src/main/java/com/baseapplication/database/JsonTypeConverter.kt
package com.baseapplication.database

import androidx.room.TypeConverter
import java.nio.charset.StandardCharsets

object JsonTypeConverter {
    @TypeConverter
    fun encodeJsonResponse(value: String): ByteArray {
        return value.toByteArray()
    }

    @TypeConverter
    fun decodeJsonResponse(encodedValue: ByteArray): String {
        return String((encodedValue), StandardCharsets.UTF_8)
    }
}
